src/main/app/document/DocumentManagementController.ts

```diff
@@ -224,133 +296,128 @@ export class DocumentManagerController extends PostController<AnyObject> {
     return refinedUrl.substring(refinedUrl.length - UID_LENGTH);
   }
 
-  public async delete(req: AppRequest<Partial<CaseWithId>>, res: Response): Promise<void> {
-    const documentsUploadedKey = 'applicantDocumentsUploaded';
-    const documentsUploaded = (req.session.userCase[documentsUploadedKey] as ListValue<UploadDocumentList>[]) ?? [];
-
-    const documentIndexToDelete = parseInt(req.params.index, 10);
-    const documentToDelete = documentsUploaded[documentIndexToDelete];
-    if (!documentToDelete?.value?.value?.citizenDocument.document_url) {
-      return res.redirect(UPLOAD_DOCUMENT);
-    }
-    const documentUrlToDelete = documentToDelete?.value?.value?.citizenDocument.document_url;
-
-    // documentsUploaded[documentIndexToDelete].value = null;
-
-    req.session.userCase = await req.locals.api.triggerEvent(
-      req.session.userCase.id,
-      { [documentsUploadedKey]: documentsUploaded },
-      CITIZEN_UPDATE
-    );
-
-    const documentManagementClient = this.getDocumentManagementClient(req.session.user);
-    await documentManagementClient.delete({ url: documentUrlToDelete });
-
-    req.session.save(err => {
-      if (err) {
-        throw err;
+  public async deleteDocument(req: AppRequest<Partial<CaseWithId>>, res: Response): Promise<void> {
+    const isApplicant = req.query.isApplicant;
+    let redirectUrl;
+    const caseworkerUser = await getSystemUser();
+    //req.session.userCase['applicantUploadFiles'];
+    const documentIdToDelete = req.params.documentId;
+    const deleteDocumentDetails = {
+      caseId: req.session.userCase.id,
+      documentId: documentIdToDelete,
+    };
+    const deleteDocumentRequest = new DeleteDocumentRequest(deleteDocumentDetails);
+    const client = new CosApiClient(caseworkerUser.accessToken, 'http://localhost:3001');
+    const deleteCitizenDocFromCos = await client.deleteCitizenStatementDocument(caseworkerUser, deleteDocumentRequest);
+    if (deleteCitizenDocFromCos === 'SUCCESS') {
+      if (isApplicant === YesOrNo.YES) {
+        req.session.userCase.applicantUploadFiles?.forEach((document, index) => {
+          if (document.id === documentIdToDelete) {
+            req.session.userCase.applicantUploadFiles?.splice(index, 1);
+          }
+        });
+      } else {
+        req.session.userCase.respondentUploadFiles?.forEach((document, index) => {
+          if (document.id === documentIdToDelete) {
+            req.session.userCase.respondentUploadFiles?.splice(index, 1);
+          }
+        });
       }
-      return res.redirect(UPLOAD_DOCUMENT);
-    });
+      const caseDataFromCos = await client.retrieveByCaseId(req.session.userCase.id, caseworkerUser);
```

Comment:
change to logged in user

Reply:
used logged in user credentials to save data in CCD